postmortems/006 - Postmortem for internet gateway route being deleted from default vpc route table.md

```diff
@@ -0,0 +1,34 @@
+# 006 - Postmortem for internet gateway route being deleted from the default vpc's route table
+
+## Level: Internal
+
+## Author(s)
+
+- Alejandro Guillen <alejandro@iasql.com>
+
+## Summary
+
+The internet gateway route have being deleted from the default vpc's route table by the `delete_all_records` method on test accounts. This has been an issue in the past for Codedeploy but something changed in AWS over the weekend (march 17th-19th) that made the failure appear on tutorials tests (Django and Prisma) blocking the whole test suite.
+
+## Timeline
+
+- **2023-03-20**: test suite degradation. Mainly Codedeploy.
+- **2023-03-21**: test suite failing almost on every run for tutorials.
+
+## Detection
+
+The tutorial tests start failing recurrently despite the retry attempts. @dfellis decided to dig in and realized it was a network issue between ECS and ECR. Since we were already trying to handle the issue for Codedeploy it makes sense to relate both issues. So we did and look for the network configuration for an account where the tutorials were failing and found the internet gateway route absence.
```

Comment:
I would link to [this](https://repost.aws/knowledge-center/ecs-unable-to-pull-secrets) that has the following text:

> The [AWS Fargate platform version 1.4.0](https://docs.aws.amazon.com/AmazonECS/latest/developerguide/platform-linux-fargate.html#platform-version-1-4) uses the task elastic network interface to pull the image and secrets. All network traffic flows through the elastic network interface within your Amazon Virtual Private Cloud (Amazon VPC).

It goes on to say that with 1.4.0 of Fargate it no longer has its own built-in link to ECR, so you need to make sure your VPC has access to the ECR repository you're using, which matches the error message in this screenshot:

![screenshot of ecr access failure](https://cdn.discordapp.com/attachments/920885549668114472/1087470750698590278/Screenshot_from_2023-03-20_15-20-35.png)

Reply:
updated